### DISCUSSION TITLE
Exit insert mode within :normal command

### DISCUSSION THREAD

**Discussion by fior-di-latte** [JetBrains/ideavim#1042] [Q&A]
The `:norm` command seems not to work if one wants to exit insert mode within it and continue with more commands in normal mode.

As proposed [here](https://stackoverflow.com/questions/4010890/vim-exit-insert-mode-with-normal-command), 
you would usually use CTRL-V + < esc > in vanilla vim.
**In Ideavim, this does not work.** Not sure if I'm doing sth wrong.

Best wishes!

**Comment by albatroez** [comment]
Hi, I found a workaround. You can go to insert mode in the editor and type in the CTRL+V + ESC which looks like this for me
![Screenshot 2025-04-17 at 13 41 29](https://github.com/user-attachments/assets/7ccb99cf-93b0-4391-bd06-37ef56c9afc8)
and then copy it in the editor and paste it when using the norm command, it displays as a space then
![Screenshot 2025-04-17 at 13 45 02](https://github.com/user-attachments/assets/feda0270-8831-45c2-ac0c-93dfc98d6236)
(the 2nd and 3rd spaces are the ctrl+v then escape sequence)
Maybe this could be improved to work like normal vim though

**Reply by citizenmatt** [reply]
Do I understand correctly that you're trying to type something like `:normal iHello<C-V><Esc>aWorld` so that `<C-V><Esc>` is replaced with a literal escape char?

Out of interest, what happens if you just use `<Esc>`?

**Reply by albatroez** [reply]
Yes, it is to exit insert mode. Clicking ESC key it exits the command-line mode (I think that's the name). Trying to type in <Esc> or <C-V><Esc> or ^[ just types those characters

**Reply by albatroez** [reply]
https://github.com/user-attachments/assets/fc3ac9b8-eda9-43d4-810d-2ef6fd89c636
